### ISSUE TITLE
[REVIEW]: RangeEnclosures.jl: A framework to bound function ranges

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#122]
**Submitting author:** @lucaferranti (<a href="http://orcid.org/0000-0001-5588-0920">Luca Ferranti</a>)
**Repository:** <a href="https://github.com/JuliaReach/RangeEnclosures.jl" target ="_blank">https://github.com/JuliaReach/RangeEnclosures.jl</a>
**Version:** v0.2.4
**Reviewer:** @dpsanders, @blegat
**Archive:** <a href="https://doi.org/10.5281/zenodo.10495948" target="_blank">10.5281/zenodo.10495948</a>

## Status

[![status](https://proceedings.juliacon.org/papers/ec833e982329eeac35354d5d32e4861c/status.svg)](https://proceedings.juliacon.org/papers/ec833e982329eeac35354d5d32e4861c)

Status badge code:

```
HTML: <a href="https://proceedings.juliacon.org/papers/ec833e982329eeac35354d5d32e4861c"><img src="https://proceedings.juliacon.org/papers/ec833e982329eeac35354d5d32e4861c/status.svg"></a>
Markdown: [![status](https://proceedings.juliacon.org/papers/ec833e982329eeac35354d5d32e4861c/status.svg)](https://proceedings.juliacon.org/papers/ec833e982329eeac35354d5d32e4861c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JuliaReach/RangeEnclosures.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@dpsanders & @blegat, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @odow know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaReach/RangeEnclosures.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@lucaferranti) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for extended abstracts respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?
### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaReach/RangeEnclosures.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@lucaferranti) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for extended abstracts respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @dpsanders, @blegat it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/JuliaCon/proceedings-review) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/JuliaCon/proceedings-review:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
Wordcount for `paper.tex` is 921

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.11 s (431.5 files/s, 41511.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
TeX                              8            266            178           2393
Julia                           19            128             69            511
Markdown                         6            119              0            323
YAML                             8             12              0            207
Lisp                             1             59              0            142
TOML                             3              5              0             50
Ruby                             1              8              4             45
CSS                              1              0              0              3
-------------------------------------------------------------------------------
SUM:                            47            597            251           3674
-------------------------------------------------------------------------------


Statistical information for the repository '8d948fed977100541e5517c8' was
gathered on 2023/01/23.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
lucaferranti                     1            57              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Luca Ferranti                57          100.0          0.0                7.02
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.29007/zzc7 is OK
- 10.1146/annurev-control-071420-081941 is OK
- 10.1137/141000671 is OK
- 10.1023/A:1021368627321 is OK
- 10.1023/B:NUMA.0000049462.70970.b6 is OK
- 10.1137/1.9780898717716 is OK
- 10.21105/jcon.00097 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by odow** [comment]
Pinging @blegat and @dpsanders to check you saw this issue.

**Comment by lucaferranti** [comment]
> Failed to discover a Statement of need section in paper

@odow is it a strict requirement to have a section named this way?

**Comment by odow** [comment]
I don't think so on the naming front, so long as there is a part in the paper (e.g., the introduction) that still meets the description.

**Comment by whedon** [comment]
:wave: @dpsanders, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @blegat, please update us on how your review is going (this is an automated reminder).

**Comment by odow** [comment]
@blegat @dpsanders are you still able to review this? If not, let me know and I can find someone else.

**Comment by mforets** [comment]
Gentle ping on the status of this submission @odow.

**Comment by odow** [comment]
I've nudged @blegat to look at this, since he'll be interested in it for some future work. But we should find someone other than @dpsanders

**Comment by dpsanders** [comment]
Apologies, I should be able to look at this.

**Comment by blegat** [comment]
Sorry for the delay, I'll do this this week

**Comment by schillic** [comment]
Thanks all! This is very unfortunate timing :sweat_smile:

The `IntervalArithmetic` devs decided to change their API and I expect another major breaking change from them in the near future (the changes are already in their main branch). I would prefer to make the paper compatible with that new version before publishing. Concretely, they will remove `IntervalBox` and instead use `AbstractVector{<:Interval}`, which we might then do in this package too. I need to discuss this with @mforets and @lucaferranti.

@blegat @dpsanders: You can still take a look at the paper. The only thing I expect to change is the API for multivariate functions. So whenever you see `IntervalBox`, replace it with `AbstractVector{<:Interval}`.

**Comment by schillic** [comment]
About the `IntervalBox` ...  it is complicated ...
But, as said above, I think you can go ahead and review. Below is some background.

We have some optional dependencies, which we load in the tests. Four of these packages would need to be updated to the new `IntervalArithmetic` API before we can test the new version. I have not seen any effort in these packages for the past weeks, and most of these packages are not maintained for more than a year. So it is unlikely that we will get there, and so the user is probably stuck with the current version if these dependencies are used.

Whether we will support newer versions of `IntervalArithmetic` is currently unclear. I do not like to upgrade `IntervalArithmetic` and then only run the tests with the old `IntervalArithmetic` version - who knows whether things still work. But I also do not like to not test functionality with the optional dependencies. Most of the code in this package is actually about that. So there is a conflict. One way out could be to test basic functionality without the optional dependencies and with the latest `IntervalArithmetic` version, and then switching to another environment for the optional dependencies (or the other way around). But I am not aware of a good way to do that. I guess we would need to have different test folders/environments and different test scripts. But overengineering is rarely a good thing :)

**Comment by dpsanders** [comment]
I think we should move the current `IntervalBox` code out into a separate package. If you are able to do this then please feel free.

**Comment by lucaferranti** [comment]
> I think we should move the current IntervalBox code out into a separate package.

@dpsanders maybe open an issue on `IntervalArithmetic` about this?

**Comment by blegat** [comment]
The package looks really useful and the paper is clear and well motivated. I only have a few minor comments:

- You cite [5, 7] saying they provide algorithms but not specifying which and then you mention the branch and bound algorithm without reference. Is the branch and bound algorithm developed in [5, 7] ?
- "dependency problem and the wrapping effect": do you have any reference for this ?
- "where the enclosure obtained with plain interval arithmetic (natural enclosure)": use `\empth{natural enclosure}` as it's the first time we encounter this term so you are defining it.
- For [9], you can use the JuliaCon 2019 ref which is more recent
- Isn't it weird that the user needs to implement a method that starts with an underscore: `_enclose` ?
- "Intervaloptimisation.jl" -> the `o` should be capitalized

**Comment by lucaferranti** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: :point_left:

**Comment by lucaferranti** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.29007/zzc7 is OK
- 10.1146/annurev-control-071420-081941 is OK
- 10.1137/141000671 is OK
- 10.1023/A:1021368627321 is OK
- 10.1023/B:NUMA.0000049462.70970.b6 is OK
- 10.1137/1.9780898717716 is OK
- 10.21105/jcon.00097 is OK
- 10.1007/978-0-387-36721-7_6 is OK

MISSING DOIs

- 10.1007/978-3-7091-6918-6_14 may be a valid DOI for title: The wrapping effect, ellipsoid arithmetic, stability and confidence regions

INVALID DOIs

- None
```

**Comment by lucaferranti** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: :point_left:

**Comment by lucaferranti** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.29007/zzc7 is OK
- 10.1146/annurev-control-071420-081941 is OK
- 10.1137/141000671 is OK
- 10.1023/A:1021368627321 is OK
- 10.1023/B:NUMA.0000049462.70970.b6 is OK
- 10.1137/1.9780898717716 is OK
- 10.21105/jcon.00097 is OK
- 10.1007/978-3-7091-6918-6_14 is OK
- 10.1007/978-0-387-36721-7_6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by lucaferranti** [comment]
@blegat we have now addressed your review comments. Here is a short summary

> You cite [5, 7] saying they provide algorithms but not specifying which and then you mention the branch and bound algorithm without reference. Is the branch and bound algorithm developed in [5, 7] ?

The purpose of the sentence in the introduction was simply to offer a generic motivation and give a more detailed algorithm - reference corresponse in section 2. To avoid confusion, we now added a survey paper on range enclosures methods and cite that in the introduction In section 2, we have a reference for every mentioned algorithm. We have also added a reference for the branch and bound.

> "dependency problem and the wrapping effect": do you have any reference for this ?

reference has been added

> "where the enclosure obtained with plain interval arithmetic (natural enclosure)": use \empth{natural enclosure} as it's the first time we encounter this term so you are defining it.

done


> For [9], you can use the JuliaCon 2019 ref which is more recent

updated

> Isn't it weird that the user needs to implement a method that starts with an underscore: _enclose ?

You are right that the `_enclose` internal method was an extra layer of complexity. The API has now been restructured so that users who want to add new algorithms can simply extend `RangeEnclosures.enclose`. The paper has beeen updated to reflect this.

> "Intervaloptimisation.jl" -> the o should be capitalized

fixed.

Please take a look at the new paper and let us know if you are happy with the new version.

**Comment by schillic** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: :point_left:

**Comment by schillic** [comment]
> I think we should move the current `IntervalBox` code out into a separate package. If you are able to do this then please feel free.

@dpsanders: We do not plan to work around the planned `IntervalBox` removal and rather freeze the `IntervalArithmetic` version as it is. This is simply because almost all backends have not even started supporting `IntervalArithmetic` v0.21, so we see little value in trying to support the next breaking version if it would not be usable.

Hence from our side the paper is stable.

**Comment by blegat** [comment]
The revision addresses all my comments, thanks. Looks good to me now.

**Comment by lucaferranti** [comment]
@blegat remember to tick the boxes in the message on top ;)

**Comment by lucaferranti** [comment]
@dpsanders do you have any updates / expected timeline for your review?

**Comment by lucaferranti** [comment]
@dpsanders any updates_

**Comment by dpsanders** [comment]
Apologies for the delay in reviewing this.

This is a nice package that brings together in a neat way various different range enclosure methods from a variety of packages in the Julia ecosystem.

Even though this package does not contain that much code, a user could have a difficult time learning to use each of the packages that it wraps, so this is very useful.

The paper is well-written and well-motivated, and the docs are concise and useful. 

A couple of comments:
- Maybe you could add an explicit example of how to extend the `enclose` function with a new method.
- It could be helpful to indicate the computational complexity of each method (although I appreciate that this is non-trivial!)
- You seem to have implemented your own branch-and-bound mechanism in the package. How does this differ from e.g. IntervalOptimization.jl?
- The [EAGO package](https://github.com/PSORLab/EAGO.jl) is a much more sophisticated interval optimization package, which could be useful to wrap as well. 

Otherwise I am happy to see this paper published in JOSS!

**Comment by lucaferranti** [comment]
Thank you for the review @dpsanders , please do make sure to also tick the boxes above :)


> You seem to have implemented your own branch-and-bound mechanism in the package. How does this differ from e.g. IntervalOptimization.jl?

The main difference is that in the Moore-Skelboe algorithm one **throws away** boxes which are proved to not contain an extremum. The implemented branch-and-bound, instead, aggregates boxes into a range extimator as you bisect.

Roughly,

- Start with a range estimation`Y` (initialized to empty interval)
- Consider the subdomain `Xi`, if you can prove the function is monotone over `Xi`, then no need to bisect, evaluate the function at the end points and update the range estimate to `Y \cup f(Xi)`.
- If you cannot prove monotonocity ($$f'(Xi)$$ contains a $$0$$), then bisect the box and go on (until the tolerance for stopping bisecting is met).

The strategy is very similar to a triditional branch-and-bound, I guess this is technically more a branch-and-accumulate

> Maybe you could add an explicit example of how to extend the enclose function with a new method.

This is done in the code snippet at the beginning of section 2.1, any thing in particular you find unclear there that we should clarify?

> The [EAGO package](https://github.com/PSORLab/EAGO.jl) is a much more sophisticated interval optimization package, which could be useful to wrap as well.

I'll look into it, thanks! I've opened an issue to discuss this

> It could be helpful to indicate the computational complexity of each method (although I appreciate that this is non-trivial!)

Indeed, it can be a bit challenging, I am also not sure with respect to what measure the complexity here. With respect to the input dimension? I would expect roughly all of those to suffer from the curse of dimensionality which is typical for inteval methods, but I'll see if we can add some insights about complexity of the algorithms in the paper or documentation.

**Comment by dpsanders** [comment]
Thanks @lucaferranti, I missed the snippet in section 2.1.
If that's not in the actual package docs, then please add it there too.

Thanks for fixing the [correctness issue](https://github.com/JuliaReach/RangeEnclosures.jl/issues/149) I reported in the repo. Has a new version of the package been released that contains that fix?

**Comment by schillic** [comment]
@dpsanders:
- Yes, the fix has been published.
- Yes, we added the snippet to the documentation as well ([here](https://juliareach.github.io/RangeEnclosures.jl/dev/tutorial/#Adding-a-new-enclosure-algorithm)).
- It seems that these complexity results are not so easy to grasp. Do you think it is crucial to include them to publish the paper? I understood this more as a suggestion.

**Comment by schillic** [comment]
ping @dpsanders

**Comment by dpsanders** [comment]
It all looks good to me, now thanks!
I've finished my review and completed the checklist, and I'm happy for this to proceed to publication. cc @odow

**Comment by schillic** [comment]
@odow: Could we push this over the finish line?

**Comment by odow** [comment]
Oops. Sorry, I was traveling in mid-december and missed this.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/JuliaCon/proceedings-papers/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00122/jcon.00122/10.21105.jcon.00122.pdf) :page_facing_up: :point_left:

**Comment by odow** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by odow** [comment]
@whedon set v0.2.3 as version

**Comment by whedon** [comment]
OK. v0.2.3 is the version.

**Comment by odow** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.29007/zzc7 is OK
- 10.1146/annurev-control-071420-081941 is OK
- 10.1137/141000671 is OK
- 10.1023/A:1021368627321 is OK
- 10.1023/B:NUMA.0000049462.70970.b6 is OK
- 10.1137/1.9780898717716 is OK
- 10.21105/jcon.00097 is OK
- 10.1007/978-3-7091-6918-6_14 is OK
- 10.1007/978-0-387-36721-7_6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by odow** [comment]
@schillic As a final step, can you make a Zenodo (https://zenodo.org) release and then post the DOI here? You probably need to make a new GitHub release for it as well.

**Comment by lucaferranti** [comment]
@odow the DOI is [10.5281/zenodo.10495948](https://doi.org/10.5281/zenodo.10495948)

**Comment by odow** [comment]
@whedon set v0.2.4 as version

**Comment by whedon** [comment]
OK. v0.2.4 is the version.

**Comment by odow** [comment]
@whedon set 10.5281/zenodo.10495948 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.10495948" target="_blank">10.5281/zenodo.10495948</a> is the archive.

**Comment by odow** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.29007/zzc7 is OK
- 10.1146/annurev-control-071420-081941 is OK
- 10.1137/141000671 is OK
- 10.1023/A:1021368627321 is OK
- 10.1023/B:NUMA.0000049462.70970.b6 is OK
- 10.1137/1.9780898717716 is OK
- 10.21105/jcon.00097 is OK
- 10.1007/978-3-7091-6918-6_14 is OK
- 10.1007/978-0-387-36721-7_6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @JuliaCon/jcon-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/JuliaCon/proceedings-papers/pull/76

If the paper PDF and Crossref deposit XML look good in https://github.com/JuliaCon/proceedings-papers/pull/76, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by odow** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @odow, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by odow** [comment]
cc @matbesancon

**Comment by matbesancon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JCON!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/JuliaCon/proceedings-papers/pull/77
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jcon.00122](https://doi.org/10.21105/jcon.00122)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by matbesancon** [comment]
Perfect, @lucaferranti you know the drill :)
congrats!

**Comment by schillic** [comment]
Thank you all!
Everything looks good to me, so I close this issue.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00122/status.svg)](https://doi.org/10.21105/jcon.00122)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jcon.00122">
  <img src="https://proceedings.juliacon.org/papers/10.21105/jcon.00122/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://proceedings.juliacon.org/papers/10.21105/jcon.00122/status.svg
   :target: https://doi.org/10.21105/jcon.00122
```
This is how it will look in your documentation:

[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00122/status.svg)](https://doi.org/10.21105/jcon.00122)

**We need your help!**

JuliaCon Proceedings is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://forms.gle/yANMe4c9YmuW9MWcA
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-jcon

**Comment by dpsanders** [comment]
Congrats everybody on the acceptance and the nice work :)

**Comment by lucaferranti** [comment]
following messages to test new editorial bot infrastructure

**Comment by lucaferranti** [comment]
@editorialbot commands

**Comment by lucaferranti** [comment]
@editorialbot commands

**Comment by lucaferranti** [comment]
@arfon @xuanxu it's alive :rocket:
